### ISSUE TITLE
document/cleanup module dependancy handling

### ISSUE THREAD

**Issue by samtygier-stfc** [mantidproject/mantidimaging#855]
### Desired Behaviour

Documentation that explains how to add a new module. Identify if the process can be simplified to reduce redundancy. If not check that everything is consent.

### Current Behaviour

When adding a new dependency to mantid imaging is not clear what needs to be updated.

Dependencies are listed in the following places,

- environment.yml
- deps/dev-requirements.pip
- deps/build-requirements.conda
- conda/meta.yaml
- install.sh
- setup.py

**Comment by samtygier-stfc** [comment]
**Use cases:**

### Case 1 - User

Should be about to run a simple command to install MI using conda. Ideally conda should know all the requirements, so that installing the mantidimaging package is sufficient, but not possible if some packages from pypi/pip needed.

Currently download and run install.sh, which effectively makes use of conda/meta.yaml and manually adds some pip packages.


### Case 2 - User IDaaS

Needs reliable non-interactive commands. Needs to handle release and nightly, which may have different requirements

Ansible uses install.sh (from specific branches) to install package. 

### Case 3 - Developer

Should be able to clone the git repo, and run a simple command to get dev and run requirements.

Currently clone, then `conda env create -f environment.yml`

### Case 4 - Automated testing

Conda github action
Basically same as developer case, clone, environment.yml

Docker github action
Installs unstable branch using the install.sh, then
`pip install -r /opt/mantidimaging-deps/dev-requirements.pip`

### Case 5 - Building conda release

Currently some targets in the Makefile that use deps/build-requirements.conda

or in github action, installs conda-build conda-verify and builds. Build step uses conda/meta.yaml

**Comment by samtygier-stfc** [comment]
In an ideal situation it would "conda install mantidimaging" would be sufficient for a user install. In practise it is not because additional conda channels and some packages from pypi are needed. This means it is not possible to specify everything in conda/meta.yaml . However it is possible to specify them in an environment.yml

Proposal:

All runtime conda dependencies get specified in conda/meta.yaml . environment.yml changed to depend on only mantidimaging and pypi runtime modules. environment.yml points to nightly, fix to version in release branch.

User/idaas install becomes:

`conda env create -n mantidimaging_nightly -f https://raw.githubusercontent.com/mantidproject/mantidimaging/master/environment.yml`

`conda env create -f https://raw.githubusercontent.com/mantidproject/mantidimaging/release-2.0.0/environment.yml`


Developer install:

Add an environment-dev.yml same as environment-dev.yml but without mantidimaging and dev packages added.

```
conda env create -n mantidimaging_dev -f https://raw.githubusercontent.com/mantidproject/mantidimaging/master/environment-dev.yml
conda activate mantidimaging_dev
conda install --only-deps mantidimaging
```

Automated testing. follow developer method

Manual conda building. Update Makefile rules to avoid needing build-requirements.conda

Remove deps/dev-requirements.pip, content is moved to environment-dev.yml

Remove deps/build-requirements.conda, these are already listed in conda/meta.yaml

Repetition of packages in environment.yml and conda/meta.yaml is removed

install.sh is redundant, can be removed once IDaaS and docs updated

setup.py. drop install_requires section. These are not used by the conda work flow are insufficient on their own.
